moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -398,23 +432,25 @@ robot_model::SolverAllocatorFn KinematicsPluginLoader::getLoaderFunction(const s
 
           // Allow a kinematic solver's tip links to be specified on the rosparam server as an array
           std::string ksolver_ik_links_param_name;
-          if (nh.searchParam(base_param_name + "/kinematics_solver_ik_links", ksolver_ik_links_param_name))
+          if (ksolver_params->has_parameter(base_param_name + "/kinematics_solver_ik_links"))
           {
-            XmlRpc::XmlRpcValue ksolver_ik_links;
-            if (nh.getParam(ksolver_ik_links_param_name, ksolver_ik_links))  // has custom rosparam-based tip link(s)
-            {
-              if (ksolver_ik_links.getType() != XmlRpc::XmlRpcValue::TypeArray)
-              {
-                ROS_WARN_STREAM_NAMED(LOGNAME, "rosparam '" << ksolver_ik_links_param_name << "' should be an XmlRpc "
-                                                                                              "value type 'Array'");
-              }
-              else
-              {
+            ksolver_ik_links_param_name = node->get_parameter(
+                                base_param_name + "/kinematics_solver_ik_links").get_value<std::string>();
+            // XmlRpc::XmlRpcValue ksolver_ik_links;
+            std::vector<std::string> ksolver_ik_links;
+            if(ksolver_params->has_parameter(ksolver_ik_links_param_name)){
+              ksolver_ik_links = node->get_parameter(ksolver_ik_links_param_name).get_value<std::vector<std::string>>();
+              if(node->get_parameter(ksolver_ik_links_param_name).get_type_name().find("array")){
```

Comment:
It seems like type checking methods are all over the place here. We should clean this up and note the steps as "best practices". @JafarAbdi do you want me to make an attempt?

Reply:
Yep, I'm happy to do it -- should I do it in a separate PR .? I would replace the comparison with ` rclcpp::ParameterType::PARAMETER_STRING_ARRAY`